rmf_traffic/src/rmf_traffic/DetectConflict.cpp

```diff
@@ -23,9 +23,9 @@
 
 #include "DetectConflictInternal.hpp"
 
-#include <fcl/continuous_collision.h>
-#include <fcl/ccd/motion.h>
-#include <fcl/collision.h>
+#include <fcl/narrowphase/continuous_collision.h>
```

Comment:
We can change these includes like:

```
#ifdef RMF_TRAFFIC__USING_FCL_0_6
#include <fcl/narrowphase/continuous_collision.h>
#include <fcl/math/motion/spline_motion.h>
#include <fcl/narrowphase/collision.h>

using SplineMotion = fcl::SplineMotion<double>;
using ContinuousCollisionRequest = fcl::ContinuousCollisionRequestd;
... etc ...

#else
#include <fcl/continuous_collision.h>
#include <fcl/ccd/motion.h>
#include <fcl/collision.h>

using SplineMotion = fcl::SplineMotion;
using ContinuousCollisionRequest = fcl::ContinuousCollisionRequest;
... etc ...

#endif // RMF_TRAFFIC__USING_FCL_0_6
```

We may also want to move some of the conversion functions whose APIs changed into these `#ifdef` blocks.

We'll also need to use this `#ifdef` pattern in all the other source files that use FCL. Alternatively, we can make a "[backward compatibility](https://github.com/dartsim/dart/blob/master/dart/collision/fcl/BackwardCompatibility.hpp)" header like what's done in DART and include that in all the different source files. If we take this approach, make sure to keep it as an internal header inside `src/rmf_traffic` rather than putting it in the public API in `include/rmf_traffic`.

Reply:
0c965de fcl 0.5 backwards compatibility